### PR TITLE
A bunch of cleanup and fixes for infrastructure/util/typelist.d

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

File: infrastructure/pyd/def.d [hunk 1]
```diff
@@ -139,10 +139,10 @@ struct Args(string default_modulename,
...
     }
 
-    alias Filter!(Not!IsModuleName, 
-          Filter!(Not!IsDocstring, 
-          Filter!(Not!IsPyName,
-          Filter!(Not!IsMode,
+    alias Filter!(templateNot!IsModuleName, 
+          Filter!(templateNot!IsDocstring, 
+          Filter!(templateNot!IsPyName,
+          Filter!(templateNot!IsMode,
```
Comment (John-Colvin) [inline infrastructure/pyd/def.d]: These were the only instances of `Not` that I could find

File: infrastructure/util/typelist.d [hunk 2]
```diff
@@ -1,27 +1,6 @@
...
-            static if (pred!(lst[0]))
-                alias TypeTuple!(lst[0], tail) Filter;
-            else
-                alias tail Filter;
-        }
-        else
-            alias TypeTuple!() Filter;
-    }
-}
+public import std.typetuple: Filter;
```
Comment (John-Colvin) [inline infrastructure/util/typelist.d]: No one is using <2.061 any more

File: infrastructure/util/typelist.d [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
-            static if (pred!(lst[0]))
-                alias TypeTuple!(lst[0], tail) Filter;
-            else
-                alias tail Filter;
-        }
-        else
-            alias TypeTuple!() Filter;
-    }
-}
-
```
Comment (John-Colvin) [inline infrastructure/util/typelist.d]: Everyone has >2.061 now, std.typetuple.Filter can be safely imported directly

File: infrastructure/util/typelist.d [hunk 4]
```diff
@@ -1,51 +1,53 @@
...
 template Join(string delimit, T...) {
         static if(T.length == 0) {
             enum Join = "";
         }else static if(T.length == 1) {
             enum Join =  T[0];
         }else {
-            enum Join = T[0] ~ "," ~ Join!(delimit,T[1..$]);
+            enum Join = T[0] ~ delimit ~ Join!(delimit,T[1..$]);
```
Comment (John-Colvin) [inline infrastructure/util/typelist.d]: Good job you'd only ever used it with a "," delimiter!

File: infrastructure/util/typelist.d [hunk 5]
```diff
@@ -1,51 +1,53 @@
...
 // useful for templates too?
 template Pred(string pred) {
-    template Pred(Stuff...) {
-        alias Stuff[0] A;
-        enum Pred = mixin(pred);
+    template Pred(Stuff...)
+    if(Stuff.length == 1)
+    {
+        alias A = Stuff[0];
+        alias Pred = Alias!(mixin(pred));
```
Comment (John-Colvin) [inline infrastructure/util/typelist.d]: So that the answer can be a type, not just a value.

File: infrastructure/util/typelist.d (hunk 3 above)
Comment (ariovistus) [inline infrastructure/util/typelist.d]: yay!

**Pull request merged**